### ISSUE TITLE
Strange output from version

### ISSUE THREAD

**Issue by afbjorklund** [cri-o/cri-o#3845]
Previous output:

```console
                         _             _            
            _         _ ( )           ( )           
  ___ ___  (_)  ___  (_)| |/')  _   _ | |_      __  
/' _ ` _ `\| |/' _ `\| || , <  ( ) ( )| '_`\  /'__`\
| ( ) ( ) || || ( ) || || |\`\ | (_) || |_) )(  ___/
(_) (_) (_)(_)(_) (_)(_)(_) (_)`\___/'(_,__/'`\____)
$ crio --version
crio version 1.17.1
```
Updated output:

```console
$ crio --version
WARN[0000] Encountered error detecting link mode of binary: `ldd /usr/bin/crio` failed:   (exec: "ldd": executable file not found in $PATH) 
WARN[0000] Encountered error detecting link mode of binary: `ldd /usr/bin/crio` failed:   (exec: "ldd": executable file not found in $PATH) 
crio version 
Version:    1.18.1
GoVersion:  go1.13.8
Compiler:   gc
Platform:   linux/amd64

```

I _tried_ to supply the GIT_COMMIT also with the new make variables, but that info seems missing.
That is, it is not available when building the release from the tarball but only when building from git.

Laptop output:
```
crio version 
Version:       1.18.1
GitCommit:     5cbf694c34f8d1af19eb873e39057663a4830635
GitTreeState:  clean
BuildDate:     2020-06-07T15:03:33Z
GoVersion:     go1.13.8
Compiler:      gc
Platform:      linux/amd64
Linkmode:      dynamic
```

`make COMMIT_NO=5cbf694c34f8d1af19eb873e39057663a4830635 PREFIX=/usr binaries`

Is there something more needed ? Are there other variables to supply this version information ?

```console
 file out/buildroot/output/target/usr/bin/crio
out/buildroot/output/target/usr/bin/crio: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 5.4.0, Go BuildID=pnqFhA7h8z8BERVxjB6D/80jgCVpBQ1L4XCb7mONC/tDSx0k9EPYOozoKN1jb6/vx4KwRTOsbebCXcsecT9, stripped
 ldd out/buildroot/output/target/usr/bin/crio
	linux-vdso.so.1 (0x00007fff27d94000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fa06f8b1000)
	libseccomp.so.2 => /usr/lib/x86_64-linux-gnu/libseccomp.so.2 (0x00007fa06f85e000)
	libgpgme.so.11 => /usr/lib/x86_64-linux-gnu/libgpgme.so.11 (0x00007fa06f80d000)
	libassuan.so.0 => /usr/lib/x86_64-linux-gnu/libassuan.so.0 (0x00007fa06f7f8000)
	libgpg-error.so.0 => /lib/x86_64-linux-gnu/libgpg-error.so.0 (0x00007fa06f7d5000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fa06f7ca000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fa06f5d6000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fa06f903000)
```

**Comment by afbjorklund** [comment]
As per above, we expect the output to look like:

```
$ crio --version
crio version 1.18.1
```

This new output doesn't really parse any longer.

**Comment by afbjorklund** [comment]
The reason the git information is gone, is because it is now using the global ${LDFLAGS} variable.

```Makefile
BASE_LDFLAGS = ${SHRINKFLAGS} \
	-X ${PROJECT}/internal/pkg/criocli.DefaultsPath=${DEFAULTS_PATH} \
	-X ${PROJECT}/internal/version.buildDate=$(shell date -u +'%Y-%m-%dT%H:%M:%SZ') \
	-X ${PROJECT}/internal/version.gitCommit=${COMMIT_NO} \
	-X ${PROJECT}/internal/version.gitTreeState=${GIT_TREE_STATE} \

LDFLAGS = -ldflags '${BASE_LDFLAGS} ${EXTRA_LDFLAGS}'
```

Since we build with CPPFLAGS/CFLAGS/CXXFLAGS/LDFLAGS set, the version information is lost.

`LDFLAGS=""`

This EXTRA_LDFLAGS is not really described anywhere, not more than COMMIT_NO was anyway.

**Comment by afbjorklund** [comment]
Anyway, the end result is only that the version number is missing from the start information:

`🎁  Preparing Kubernetes v1.18.2 on CRI-O  ...`

It seems to be working OK.

**Comment by haircommander** [comment]
@afbjorklund all set here?

**Comment by afbjorklund** [comment]
The only question was if it should be cherry-picked to master, as I only patched out v1.18.1 install

**Comment by haircommander** [comment]
I'd be fine cherry picking to master. can you take care of that @afbjorklund ?

**Comment by afbjorklund** [comment]
@haircommander : cherry-picked (and conflict resolved)

I would still suggest leaving the extra output for `crio version`,
but it is up to you... Seems better there, and has `--json` etc.

**Comment by afbjorklund** [comment]
Was there any problem with the uplift in PR #3886 ?

**Comment by haircommander** [comment]
> Was there any problem with the uplift in PR #3886 ?

nope, just fell through the cracks

**Comment by afbjorklund** [comment]
Fixed on master, d0dc0d3076367f6a26c46fb6e72d1e582b552599

```
crio version 1.19.0-dev
Version:       1.19.0-dev
GitCommit:     d0dc0d3076367f6a26c46fb6e72d1e582b552599
GitTreeState:  clean
...

```
